cmd/auth/add.go

```diff
@@ -109,6 +99,15 @@ var addCmd = &cobra.Command{
 			password = strings.TrimSpace(string(bytePassword))
 		}
 
+		// search for provider with same name
+		providerIndex := -1
+		for i, provider := range configAI.Providers {
+			if backend == provider.Name || defaultBackend == provider.Name {
```

Comment:
Hi @amitamrutiya2210, I was wondering if `defaultBackend == provider.Name` is still needed? I'd suggest taking a look at https://github.com/k8sgpt-ai/k8sgpt/pull/929/files#diff-1f69cd9b70b884bda8de404da6c720e50b8aa29af0ba3365563a44f532ead7a6 for clearer understanding.

Reply:
Yes, this is necessary until your PR is merged. In your PR, this problem is solved in a different manner with a lot of code changes. So, until your PR is not merged, this solution remains valid.